File: backend/src/main/scala/microservices/hotel/objects/UploadHotelRoomTypeImagePlannerRequest.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadHotelRoomTypeImagePlannerRequest(
    originalFileName: String,
    mimeType: String,
    fileContentBase64: String
)

object UploadHotelRoomTypeImagePlannerRequest:
  given sourceEncoder: Encoder[UploadHotelRoomTypeImagePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UploadHotelRoomTypeImagePlannerRequest] = deriveDecoder

File: frontend/src/microservices/hotel/objects/UploadHotelRoomTypeImagePlannerRequest.ts
export type UploadHotelRoomTypeImagePlannerRequest = {
  originalFileName: string
  mimeType: string
  fileContentBase64: string
}

export const uploadHotelRoomTypeImagePlannerRequestFromJson = (json: string): UploadHotelRoomTypeImagePlannerRequest =>
  JSON.parse(json) as UploadHotelRoomTypeImagePlannerRequest

export const uploadHotelRoomTypeImagePlannerRequestToJson = (value: UploadHotelRoomTypeImagePlannerRequest): string =>
  JSON.stringify(value)
